### PR TITLE
Hide node deprecation warning on CLI commands

### PR DESCRIPTION
**Pull request by acunniffe**
There's an open issue in the latest mockttp (recently updated) where an intermediate deprecation warning is being emitted. https://github.com/httptoolkit/mockttp/issues/41

This change hides those to clean up the DX of the CLI.

### CODE REVIEW

File: workspaces/local-cli/bin/run.cmd [hunk 1]
```diff
@@ -1,3 +1,3 @@
 @echo off
 
-node "%~dp0\run" %*
+node "%~dp0\run" %* --no-warnings
```
Comment (devdoshi) [inline workspaces/local-cli/bin/run.cmd:3]: node  --no-warnings "%~dp0\run" %*

**Comment by JaapRood** [comment]
@acunniffe @devdoshi can any of shed some light on exactly when and how this mutes warnings? Can we be sure that we _do_ get warnings in local development?

**Comment by LouManglass** [comment]
> can any of shed some light on exactly when and how this mutes warnings? Can we be sure that we _do_ get warnings in local development? Isn't `--no-deprecation` supposed to be [a thing](https://nodejs.org/api/cli.html#cli_no_deprecation) instead of `--no-warnings`?

In my testing, local development invocations do not return deprecation warnings. They're hidden in anything I've tried.

**Comment by devdoshi** [comment]
There was an issue with this approach (see https://github.com/opticdev/optic/commit/9c5cd2d96445c5ffeecc6f6e62c4384e90fad574) but with that being the code you're running against, you can see why you won't see warnings. Easy to put it behind an environment variable, or use our developerDebugLogger, or ... for development, so let me know your thoughts @LouManglass @JaapRood

**Comment by JaapRood** [comment]
Development should never hide warnings, that's what the warnings are there for: to make sure we move ourselves out of harms way and don't create way bigger issues down the line. It might be safe to ignore this particular warning, but there is 0 guarantee that's the case for the next one. Anything we merged that doesn't account for that should be reverted until that's addressed.

**Pull request merged**